app/src/main/java/com/rkm/tasky/feature/loginscreen/screen/LoginScreen.kt

```diff
@@ -0,0 +1,157 @@
+package com.rkm.tasky.feature.loginscreen.screen
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import com.rkm.tasky.R
+import com.rkm.tasky.feature.loginscreen.viewmodel.LoginViewModel
+import com.rkm.tasky.ui.component.EmailValidationTextField
+import com.rkm.tasky.ui.component.PasswordTextField
+
+@Composable
+fun LoginScreen(
+    viewModel: LoginViewModel,
+    modifier: Modifier
+) {
+
+    // TODO: Change to collectAsStateWithLifeCycle 
+
+    val email by viewModel.emailState.collectAsState()
+    val isValidEmail by viewModel.isValidEmailState.collectAsState()
+    val password by viewModel.passwordState.collectAsState()
+    val showPassword by viewModel.showPasswordState.collectAsState()
+
+    ConstraintLayout(
+        modifier = modifier
+            .fillMaxSize()
+            .background(Color.Black)
+    ) {
+
+        val (greetingText, loginBodyLayout) = createRefs()
+        val textGuideLine = createGuidelineFromTop(0.15f)
+        val constraintLayoutGuideline = createGuidelineFromTop(0.3f)
```

Comment:
When creating UI in Compose, you´re good with Columns, Rows and Boxes in 99% of the cases. I prefer to opt for those building blocks as I've found it aids in modularizing my UI code by creating smaller, reusable components and it makes the UI simpler to visualize when you don't have a preview available. ConstraintLayout might only be useful if you have to animate those constraints, but in all other cases I´d go for the other layouts.

I'm not saying you shouldn't use `ConstraintLayout`, heck it was my bread and butter when writing XML views 😄, but just sharing my preference and reasoning.